### PR TITLE
feat: Add readme for fast transformers huggingface bert

### PR DESCRIPTION
**Pull request by jagadeeshi2i**
## Description

Adds readme doc for fast transformers huggingface bert example.

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the tests [UT/IT] that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced. 
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- UT/IT execution results

- Logs

## Checklist:

- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] New and existing unit tests pass locally with these changes?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: kubernetes/examples/FasterTransformer_HuggingFace_Bert.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
+# Faster Transformer HuggingFace Bert example in Kubernetes Torchserve.
```
Comment (msaroufim) [inline kubernetes/examples/FasterTransformer_HuggingFace_Bert.md:1]: Hamid already has an example with a similar name https://github.com/pytorch/serve/tree/master/examples/FasterTransformer_HuggingFace_Bert

Perhaps we can rename this tutorial HuggingFace Faster Transformer Kubernetes or expand on the tutorial @HamidShojanazeri  has

File: kubernetes/examples/FasterTransformer_HuggingFace_Bert.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+Batch inferencing with Transformers faces two challenges
+
+    Large batch sizes suffer from higher latency and small or medium-sized batches this will become kernel latency launch bound.
+    Padding wastes a lot of compute, (batchsize, seq_length) requires to pad the sequence to (batchsize, max_length) where difference between avg_length and max_length results in a considerable waste of computation, increasing the batch size worsen this situation.
+
+Faster Transformers (FT) from Nvidia along with Efficient Transformers (EFFT) that is built on top of FT address the above two challenges, by fusing the CUDA kernels and dynamically removing padding during computations. The current implementation from Faster Transformers support BERT like encoder and decoder layers. In this example, we show how to get a Torchsctipted (traced) EFFT variant of Bert models from HuggingFace (HF) for sequence classification and question answering and serve it.
```
Comment (msaroufim) [inline kubernetes/examples/FasterTransformer_HuggingFace_Bert.md:8]: Mention HF version limitation

File: kubernetes/examples/FasterTransformer_HuggingFace_Bert.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+[GKE Cluster setup](../GKE/README.md##-TorchServe-on-Google-Kubernetes-Engine-GKE)
+
+Once the cluster and the PVCs are ready, we can generate MAR file.
+
+## Generate Mar file
+
+1. Start a pytorch docker container
+
+```bash
+nvidia-docker run --gpus all -it nvcr.io/nvidia/pytorch:21.08-py3 bash &
```
Comment (msaroufim) [inline kubernetes/examples/FasterTransformer_HuggingFace_Bert.md:29]: does this docker container change? If so add a link to where it gets updated

File: kubernetes/examples/FasterTransformer_HuggingFace_Bert.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+cd examples/Huggingface_Transformers
+python Download_Transformer_models.py
+
+cd /workspace/FasterTransformer/build/
+
+python pytorch/Bert_FT_trace.py --mode sequence_classification --model_name_or_path "/workspace/serve/examples/Huggingface_Transformers/Transformer_model" --tokenizer_name "bert-base-uncased" --batch_size 1 --data_type fp16 --model_type thsext
+
+cd /workspace/serve/examples/Huggingface_Transformers
+
+# make sure to change the ../Huggingface_Transformers/setup_config.json "save_mode":"torchscript" and "FasterTransformer":true
```
Comment (msaroufim) [inline kubernetes/examples/FasterTransformer_HuggingFace_Bert.md:92]: just put the changed setup_config here

File: kubernetes/examples/FasterTransformer_HuggingFace_Bert.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+
+# change the ../Huggingface_Transformers/setup_config.json
+
+{
+"model_name":"bert-base-uncased",
+"mode":"sequence_classification",
+"do_lower_case":true,
+"num_labels":"0",
+"save_mode":"pretrained",
```
Comment (msaroufim) [inline kubernetes/examples/FasterTransformer_HuggingFace_Bert.md:101]: torchscript

File: kubernetes/examples/FasterTransformer_HuggingFace_Bert.md [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+
+```bash
+git clone https://github.com/pytorch/serve.git
+cd serve/docker
+```
+
+2. Modify Python and Pip paths in `Dockerfile` as below
+
+```bash
+sed -i 's#/usr/bin/python3#/opt/conda/bin/python3#g' Dockerfile
```
Comment (msaroufim) [inline kubernetes/examples/FasterTransformer_HuggingFace_Bert.md:164]: Is this a must?

File: kubernetes/examples/FasterTransformer_HuggingFace_Bert.md [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+
+```bash
+time_namelookup:    0.000021s
+time_connect:       0.000078s
+time_appconnect:    0.000000s
+time_pretransfer:   0.000096s
+time_redirect:      0.000000s
+time_starttransfer: 0.000922s
+                    ----------
+time_total: 0.007053s
```
Comment (msaroufim) [inline kubernetes/examples/FasterTransformer_HuggingFace_Bert.md:278]: Does this measurement include preprocessing, inference and post processing from the handler? I don't have much experience with K8 but would be curious more to learn about how stuff is benchmarked here

**Review by msaroufim — CHANGES_REQUESTED** [review]

File: kubernetes/examples/FasterTransformer_HuggingFace_Bert.md (hunk 2 above)
Comment (chauhang) [reply kubernetes/examples/FasterTransformer_HuggingFace_Bert.md:8]: @jagadeeshi2i No need to repeat these details here again. You can point to the FasterTransformer example [link](https://github.com/pytorch/serve/tree/master/examples/FasterTransformer_HuggingFace_Bert) and describe the parts related to K8s deployments for it in this readme

File: kubernetes/examples/FasterTransformer_HuggingFace_Bert.md (hunk 6 above)
Comment (jagadeeshi2i) [reply kubernetes/examples/FasterTransformer_HuggingFace_Bert.md:164]: Yes the ngv pytorch containers use python from conda.

File: kubernetes/examples/FasterTransformer_HuggingFace_Bert.md (hunk 7 above)
Comment (jagadeeshi2i) [reply kubernetes/examples/FasterTransformer_HuggingFace_Bert.md:278]: This just give a high level info.

File: kubernetes/examples/FasterTransformer_HuggingFace_Bert.md (hunk 4 above)
Comment (jagadeeshi2i) [reply kubernetes/examples/FasterTransformer_HuggingFace_Bert.md:92]: Linking to Faster Transformers readme

**Review by HamidShojanazeri — APPROVED** [review]

**Review by msaroufim — APPROVED** [review]

**Pull request merged**